guardian/floodgate
# Modernisation PR #1 - CDK, etc.

## What does this change?

- Implements Github actions builds
- Implements CDK infrastructure
- Infrastructure now deploys via Riffraff
- Implements central ELK logging

**Note** the Dynamo tables are now cloudformed as well, within the CDK. This means that any existing data must be backed up out of the existing tables and copied back in again when this is first deployed.

## How to test

Use the "Dangerous" deployment mode to deploy a CODE stack directly out of RiffRaff, without a pre-existing stack.  Everything should "just work". You will be able to find logs at https://logs.gutools.co.uk/s/content-platforms/app/discover#/view/bb4f8350-7ad0-11ed-a063-4909f0248ecd?_g=(filters:!(),refreshInterval:(pause:!t,value:10000),time:(from:now%2Fd,to:now%2Fd))&_a=(columns:!(stage,logger_name,message,level,ec2_instance_id),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:b0be43a0-59d7-11e8-a75a-b7af20e8f748,key:app,negate:!f,params:(query:content-api-floodgate),type:phrase),query:(match_phrase:(app:content-api-floodgate)))),grid:(),hideChart:!f,index:b0be43a0-59d7-11e8-a75a-b7af20e8f748,interval:auto,query:(language:lucene,query:'*'),sort:!(!('@timestamp',desc))).

Note that CODE will have no sources set up as these will need to be done from the dynamodb tables.

## How can we measure success?

- Simpler deployment
- Better logging

## Have we considered potential risks?

n/a
